source/particles/property_pool.cc

```diff
@@ -167,6 +167,60 @@ namespace Particles
     return n_properties;
   }
 
+  template <int dim, int spacedim>
+  void
+  PropertyPool<dim, spacedim>::sort_memory_slots(
+    std::vector<std::vector<Handle>> &sorted_handles)
+  {
+    std::vector<Point<spacedim>> sorted_locations(locations.size());
+    std::vector<Point<dim>>      sorted_reference_locations(
+      reference_locations.size());
+    std::vector<types::particle_index> sorted_ids(ids.size());
+    std::vector<double>                sorted_properties(properties.size());
```

Comment:
Would it make sense to only reserve the memory and then call `push_back()` below, in order to avoid a second pass through memory?
```suggestion
    std::vector<Point<spacedim>> sorted_locations;
    sorted_locations.reserve(locations.size());
    std::vector<Point<dim>>      sorted_reference_locations;
    sorted_reference_locations.reserve(reference_locations.size());
    std::vector<types::particle_index> sorted_ids;
    sorted_ids.reserve(ids.size());
    std::vector<double> sorted_properties;
    sorted_properties.reserve(properties.size());
```